File: 1396-count-servers-that-communicate/count-servers-that-communicate.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        rowLookup = {}
        columnLookup = {}
        rows = len(grid)
        columns = len(grid[0])
        count = 0

        for i in range(rows):
            for j in range(columns):
                cell = grid[i][j]

                if (cell):
                    if i not in rowLookup:
                        rowLookup[i] = 0
                    rowLookup[i] += 1

                    if j not in columnLookup:
                        columnLookup[j] = 0
                    columnLookup[j] += 1

        for i in range(rows):
            for j in range(columns):
                cell = grid[i][j]

                if (cell):
                    if rowLookup[i] > 1 or columnLookup[j] > 1:
                        count += 1

        return count
                
        

File: 1396-count-servers-that-communicate/count-servers-that-communicate.js
var countServers = function(grid) {
    let rows = grid.length
    let columns = grid[0].length
    let rowCount = {}
    let columnCount = {}
    let count = 0

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            let cell = grid[i][j]
            if (cell) {
                if (!rowCount[i]) rowCount[i] = 0
                rowCount[i] ++
                if (!columnCount[j]) columnCount[j] = 0
                columnCount[j] ++
            }
        }
    }    

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            let cell = grid[i][j]
            if (cell) {
                if (rowCount[i] > 1 || columnCount[j] > 1) count += 1
            }
        }
    }    
    return count


};